src/test/java/com/test/combopack/packer/DynamicPackagerTest.java

```diff
@@ -1,89 +1,36 @@
 package com.test.combopack.packer;
 
-import static com.pack.combopack.util.PackageUtility.getTotalWeight;
 import static com.test.combopack.moc.ItemProvider.availableBoxs;
-import static com.test.combopack.moc.ItemProvider.getAvailableItems;
-import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
-import java.util.ArrayList;
 import java.util.List;
 
+import org.junit.Before;
 import org.junit.Test;
 
 import com.pack.combopack.bean.Box;
 import com.pack.combopack.bean.Item;
-import com.pack.combopack.exception.PackagingException;
+import com.pack.combopack.bean.Packable;
 import com.pack.combopack.packer.DynamicPackager;
-import com.pack.combopack.packer.KnapsackPackager;
 
-public class DynamicPackagerTest {
+public class DynamicPackagerTest extends AbstractKnapSackPackagerTest {
 
-	KnapsackPackager packer = new DynamicPackager();
+    @Before
+    public void init() {
+        packer = new DynamicPackager();
+    }
 
-	@Test
-	public void pack() throws PackagingException {
+    @Test
+    public void pack() {
 
-		for (Box<Item> box : availableBoxs) {
-			List<Item> items = packer.pack(getAvailableItems(box.getId()),
-					box.getMaxWeight());
-			assertTrue(getTotalWeight(items) < box.getMaxWeight());
-			box.setItems(items);
-			System.out.println(box);
-		}
-	}
+        System.out.println("\n <!-- Packaging with : " + packer.getClass().getSimpleName() + "-->");
 
-	@Test(expected = PackagingException.class)
-	public void packWithNULLItems() throws PackagingException {
-		packer.pack(null, 1.0d);
+        for (Box<Item> box : availableBoxs) {
 
-	}
+            List<Packable> items = packer.pack(box);
+            assertTrue(packer.getTotalWeight(items) < box.getCapacity());
```

Comment:
**Scalability Issue**: The method packer.getTotalWeight(items) is used within assertTrue in tests without considering large datasets, which might not scale well. <br> **Fix**: Implement a mock or stub for packer.getTotalWeight(items) that can handle large datasets efficiently, possibly by using a scalable algorithm or data structure. <br> **Code Suggestion**: 
 ```
 -            assertTrue(packer.getTotalWeight(items) < box.getCapacity());
 +            // Mock or stub implementation for scalability
 +            assertTrue(mockTotalWeight(items) < box.getCapacity());
 ```